### ISSUE TITLE
[critical] Composed speech SFT: loss not normalized for gradient accumulation (`num_items_in_batch` swallowed)

### ISSUE THREAD

**Issue by nagohachi** [nagohachi/glide#3]
## Defect

`SpeechLLM.forward` accepts `**kwargs` and deliberately swallows `num_items_in_batch` ([speech_llm.py#L130-L134](https://github.com/nagohachi/glide/blob/1cacf15da0d5e976c3db81648c829ee15535ab1a/src/glide/models/speech_llm.py#L130-L134)):

```python
def forward(self, input_ids=None, attention_mask=None, labels=None, ..., **kwargs):
    # Ignore extra Trainer kwargs (use_cache, num_items_in_batch, ...)
```

But the `**kwargs` (VAR_KEYWORD) signature makes HF `Trainer` set `model_accepts_loss_kwargs=True`. Consequences in `transformers` (verified against 5.12.1, same logic since 4.46):

1. `training_step` **skips the `loss / gradient_accumulation_steps` division** — but the inner LLM returned a *mean* CE loss, so gradients are ~GA× too large.
2. With DDP and `average_tokens_across_devices=True` (the default), `compute_loss` additionally multiplies the loss by `num_processes`.

`configs/base.yaml` defaults `gradient_accumulation_steps: 8`, so every composed speech SFT run trains at an effective LR ~8× (single GPU) to ~8×N× (DDP) the configured value.

`rl_speech.py` is **not** affected (its batch is a list, so `num_items_in_batch` stays `None` and its custom `training_step` divides manually).

## Fix options

- Set `self.model_accepts_loss_kwargs = False` in `LengthGroupedSFTTrainer.__init__` (HF's own docstring recommends exactly this for mean-loss models), or
- Make `SpeechLLM.forward` accept `num_items_in_batch` and recompute the CE as a sum-loss / token count.

Found in a code audit (traced through `transformers` `trainer.py` `training_step`/`compute_loss`).
